### PR TITLE
fix: Async callback for langchain

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

### CODE REVIEW

**Comment by yihong0618** [comment]
@dongjing007 can you also give a look?

File: xiaogpt/langchain/callbacks.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        parent_run_id: UUID | None = None,
+        tags: list[str] | None = None,
+        metadata: dict[str, Any] | None = None,
+        **kwargs: Any,
+    ) -> None:
+        self.done.clear()
+
+    async def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
+        if token is not None and token != "":
+            print(token, end="")
```
[the remark this answers is not in the document]
Comment (frostming) [reply xiaogpt/langchain/callbacks.py:36]: Haha, that is to display the text to console in stream mode. Intended.

**Comment by dongjing007** [comment]
> @dongjing007 can you also give a look?

ok

File: xiaogpt/langchain/callbacks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yihong0618) [reply xiaogpt/langchain/callbacks.py:36]: copy that!

**Comment by dongjing007** [comment]
Asynchronous solutions are even more elegant and praiseworthy. 👍 @frostming  @yihong0618

**Comment by frostming** [comment]
> Asynchronous solutions are even more elegant and praiseworthy. 👍 @frostming @yihong0618

Yeah, we don't have to wait for the chain run complete to get the TTS.

**Pull request merged**